migrate/v0_11/migrate.go

```diff
@@ -65,3 +67,53 @@ func MigrateBep3(oldGenState v0_9bep3.GenesisState) v0_11bep3.GenesisState {
 		PreviousBlockTime: v0_11bep3.DefaultPreviousBlockTime,
 	}
 }
+
+// MigrateCDP migrates from a v0.9 (or v0.10) cdp genesis state to a v0.11 cdp genesis state
+func MigrateCDP(oldGenState v0_9cdp.GenesisState) v0_11cdp.GenesisState {
+	var newCDPs v0_11cdp.CDPs
+	var newDeposits v0_11cdp.Deposits
+	var newCollateralParams v0_11cdp.CollateralParams
+	newStartingID := uint64(0)
+
+	for _, cdp := range oldGenState.CDPs {
+		newCDP := v0_11cdp.NewCDPWithFees(cdp.ID, cdp.Owner, cdp.Collateral, "bnb-a", cdp.Principal, cdp.AccumulatedFees, cdp.FeesUpdated)
+		newCDPs = append(newCDPs, newCDP)
+		if cdp.ID >= newStartingID {
+			newStartingID = cdp.ID + 1
+		}
+	}
+
+	for _, dep := range oldGenState.Deposits {
+		newDep := v0_11cdp.NewDeposit(dep.CdpID, dep.Depositor, dep.Amount)
+		newDeposits = append(newDeposits, newDep)
+	}
+
+	for _, cp := range oldGenState.Params.CollateralParams {
+		newCollateralParam := v0_11cdp.NewCollateralParam(cp.Denom, "bnb-a", cp.LiquidationRatio, cp.DebtLimit, cp.StabilityFee, cp.AuctionSize, cp.LiquidationPenalty, 0x01, cp.SpotMarketID, cp.LiquidationMarketID, cp.ConversionFactor)
+		newCollateralParams = append(newCollateralParams, newCollateralParam)
+	}
+	btcbCollateralParam := v0_11cdp.NewCollateralParam("btcb", "btcb-a", sdk.MustNewDecFromStr("1.5"), sdk.NewCoin("usdx", sdk.NewInt(100000000000)), sdk.MustNewDecFromStr("1.000000001547125958"), sdk.NewInt(100000000), sdk.MustNewDecFromStr("0.075000000000000000"), 0x02, "btc:usd", "btc:usd:30", sdk.NewInt(8))
```

Comment:
I think that the market denom should match the on-chain asset denom here (applies to `xrp:usd` as well). Theoretically, `btcb` and `btc` could experience price variation which could cause issues if a user can transfer in `btcb` worth $10,000 and use it to take out a loan on Kava at a `btc` valuation of $12,000.

We've priced `busd` at `busd:usd` market prices for this reason - if `busd `is off its peg and is worth $1.02 but we treat it as USD always worth $1 then it creates an illegitimate arbitrage opportunity for users. Shouldn't `btcb`/`xrpb` be treated the same way?

Reply:
There is no market liquid enough to try and do individual price feeds for the various pegged assets, and thus we assume that btc:btcb ~= 1. Trying to overcome this risk with oracles is risky, because all of the markets where btcb is traded are very thin - thus manipulating the price of the oracle is pretty cheap. FWIW, Maker and Compound assume that pegged assets trade 1:1 as well. https://compound.finance/docs/prices (ie, BTC markets on coinbase pro are used to determine the price of WBTC).